### PR TITLE
Improve resolve exception

### LINKED ISSUE #1422: Improve error message on unresolved dependencies
## steps

Something missing in transitive dependency due to people deleting artifacts from artifactory or depending on SNAPSHOT or whatnot. 
## problem

@paulp wrote in #381:

>  Dammit Jim, I'm a programmer, not a hexadecimal scanner.

```
[info] Loading project definition from /scratch/trunk1/project/project
[info] Updating {file:/scratch/trunk1/project/project/}default-2ac879...
... snip 38 lines of resolving ...
[info] Done updating.
Cloning into '/Users/paulp/.sbt/0.12.0-M1/staging/eff7d7476f58ab859d23'...
[info] Loading project definition from /Users/paulp/.sbt/0.12.0-M1/staging/eff7d7476f58ab859d23/project
[info] Updating {file:/Users/paulp/.sbt/0.12.0-M1/staging/eff7d7476f58ab859d23/project/}default-9f14df...
[info] Resolving org.scala-tools.sbt#scripted-plugin_2.9.1;0.12.0-M1 ...
[warn]  module not found: org.scala-tools.sbt#scripted-plugin_2.9.1;0.12.0-M1

... blah blah lots more resolving
[warn]  ::::::::::::::::::::::::::::::::::::::::::::::
[warn]  ::          UNRESOLVED DEPENDENCIES         ::
[warn]  ::::::::::::::::::::::::::::::::::::::::::::::
[warn]  :: org.scala-tools.sbt#scripted-plugin_2.9.1;0.12.0-M1: not found
[warn]  ::::::::::::::::::::::::::::::::::::::::::::::
[error] {file:/Users/paulp/.sbt/0.12.0-M1/staging/eff7d7476f58ab859d23/project/}default-9f14df/*:update: sbt.ResolveException: unresolved dependency: org.scala-tools.sbt#scripted-plugin_2.9.1;0.12.0-M1: not found
```
## expectation

sbt should provide as much info on where the missing dependency is coming from in plain English, and possibly advise the user on what to do/read/check next.

### PR DESCRIPTION
**Pull request by eed3si9n**
fixes #1422, #381 
## steps

``` scala
lazy val d = project.
  settings(
    libraryDependencies += "org.apache.cayenne.plugins" % "maven-cayenne-plugin" % "3.0.2"
  )
```
## problem

You have no idea which transitive deps path failed.
## what this changes

```
[warn]  ::::::::::::::::::::::::::::::::::::::::::::::
[warn]  ::          UNRESOLVED DEPENDENCIES         ::
[warn]  ::::::::::::::::::::::::::::::::::::::::::::::
[warn]  :: foundrylogic.vpp#vpp;2.2.1: not found
[warn]  ::::::::::::::::::::::::::::::::::::::::::::::
[warn] 
[warn]  Note: Unresolved dependencies path:
[warn]      foundrylogic.vpp:vpp:2.2.1
[warn]        +- org.apache.cayenne:cayenne-tools:3.0.2
[warn]        +- org.apache.cayenne.plugins:maven-cayenne-plugin:3.0.2
[warn]        +- d:d_2.10:0.1-SNAPSHOT
```

### CODE REVIEW

File: ivy/src/main/scala/sbt/IvyActions.scala [hunk 1]
```diff
@@ -209,8 +220,15 @@ object IvyActions {
       val err =
         if (resolveReport.hasError) {
           val messages = resolveReport.getAllProblemMessages.toArray.map(_.toString).distinct
-          val failed = resolveReport.getUnresolvedDependencies.map(node => IvyRetrieve.toModuleID(node.getId))
-          Some(new ResolveException(messages, failed))
+          val failedPaths = Map(resolveReport.getUnresolvedDependencies map { node =>
+            val m = IvyRetrieve.toModuleID(node.getId)
```
Comment (jsuereth) [inline ivy/src/main/scala/sbt/IvyActions.scala:224]: Is this the part you worried was slow?

File: ivy/src/main/scala/sbt/IvyRetrieve.scala [hunk 2]
```diff
@@ -51,4 +51,34 @@ object IvyRetrieve {
     new UpdateStats(report.getResolveTime, report.getDownloadTime, report.getDownloadSize, false)
   def configurationReport(confReport: ConfigurationResolveReport): ConfigurationReport =
     new ConfigurationReport(confReport.getConfiguration, moduleReports(confReport), evicted(confReport))
+
+  /**
+   * Tries to find Ivy graph path the from node to target.
+   */
+  def findPath(target: IvyNode, from: ModuleRevisionId): List[IvyNode] = {
```
Comment (jsuereth) [inline ivy/src/main/scala/sbt/IvyRetrieve.scala:58]: Ah, so this might be faster if you grab all the paths at the same time, and memoize parent relationships.

Otherwise, I'm fine being slow + dirty :)

**Comment by jsuereth** [comment]
LGTM

File: ivy/src/main/scala/sbt/IvyActions.scala (hunk 1 above)
Comment (eed3si9n) [inline ivy/src/main/scala/sbt/IvyActions.scala:224]: I'm guessing you picked it up when you saw `findPath`, but the imple of `findPath` is what I'm worried.
